File: commons/dto/src/main/java/net/blueshell/dto/EventSignUpDTO.java
package net.blueshell.dto;

import jakarta.validation.Valid;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;


@Data
@EqualsAndHashCode(callSuper = false)
public class EventSignUpDTO extends BaseDTO {
    private Long id;
    private Long eventId;
    private String fullName;
    private String discord;
    private String email;
    @Valid
    private List<Object> formAnswers;
}

File: api/src/main/kotlin/net/blueshell/api/event/web/dto/EventSignUpDTO.kt
package net.blueshell.api.event.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import net.blueshell.api.shared.dto.AuditedAutoIdDTO
import net.blueshell.api.survey.web.dto.AnswerDTO
import net.blueshell.api.user.web.dto.SimpleUserDTO

@Schema(name = "EventSignUp")
@net.blueshell.api.event.web.validation.ValidEventSignUp
@net.blueshell.api.event.web.validation.GuestOrUserRequired
data class EventSignUpDTO(
    @field:NotNull
    var eventId: Long? = null,

    @field:Valid
    var answers: MutableList<AnswerDTO> = mutableListOf(),

    var guest: GuestDTO? = null,
    var user: SimpleUserDTO? = null,
    var userId: Long? = null
) : AuditedAutoIdDTO()
